### ISSUE TITLE
[a11y]:  dropdown, overflow menu and ghost button text contrast issues with "increase contrast" setting on

### ISSUE THREAD

**Issue by johanna-koval** [carbon-design-system/carbon#10844]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

10.42

### React version

7.43.0

### Automated testing tool and ruleset

Accessibility Checker, WCAG 1.4.3: Text Contrast

### Assistive technology

Increase contrast MacOS accessibility setting

### Description

The focus stage for **overflow menu** options, **ghost button** and inline **dropdown menu** options is against the 1.4.3 Contrast (Minimum) when the accessibility setting "increase contrast" on MacOS is on.
<img width="1213" alt="CleanShot 2022-02-23 at 15 59 06@2x" src="https://user-images.githubusercontent.com/5809654/155747885-a99a7de6-ece4-4bee-9202-0d3e5effe7bd.png">
<img width="1243" alt="CleanShot 2022-02-23 at 15 50 55@2x" src="https://user-images.githubusercontent.com/5809654/155747899-aae25893-e1db-42ba-a2c1-0d963599191c.png">
<img width="797" alt="79393000-94c2-11ec-8715-13305a1f4277" src="https://user-images.githubusercontent.com/5809654/155748122-f6b91f17-bf59-4b5c-b9eb-6ed453974a69.png">
 

### WCAG 2.1 Violation

1.4.3 Contrast

### CodeSandbox example

-

### Steps to reproduce

- Enable "increase contrast" on MacOS accessibility settings
- Using the Tab key to move until you see the focus state of the ghost button or option in an overflow menu or inline dropdown

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
